pkg/certificateauthority/handler.go

```diff
@@ -0,0 +1,64 @@
+package certificateauthority
+
+import (
+	"context"
+	"os"
+
+	"github.com/sirupsen/logrus"
+	corev1 "k8s.io/api/core/v1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+// NewHandler returns a new Handler.
+func NewHandler(client client.Client) Handler {
+	return &configmapHandler{
+		client: client,
+	}
+}
+
+// Handler is the interface that wraps the Handle method
+//
+// Handle handles a new event associated with OperatorSource type.
+//
+// ctx represents the parent context.
+// event encapsulates the event fired by operator sdk.
+type Handler interface {
+	Handle(ctx context.Context, operatorSource *corev1.ConfigMap) error
+}
+
+// configmapHandler implements the Handler interface
+type configmapHandler struct {
+	client client.Client
+}
+
+// Handle handles events associated with the ConfigMap type.
+func (h *configmapHandler) Handle(ctx context.Context, in *corev1.ConfigMap) error {
+	log := logrus.WithFields(logrus.Fields{
+		"type": in.TypeMeta.Kind,
+		"name": in.GetName(),
+	})
+
+	// Retrieve the Certificate Authority bundle from the ConfigMap.
+	caBundle := in.Data[caBundleKey]
+
+	// Retrieve the Certificate Authority bundle from Disk.
+	// If an error is returned and is not related to a nonexistant file, return an error.
+	caOnDisk, err := getCaOnDisk()
+	if err != nil && !os.IsNotExist(err) {
+		log.Infof("[ca] Error reading from disk: %v", err)
+		return err
+	}
+
+	// Compare the Certificate Authority bundle on disk with the one in the ConfigMap.
+	if caBundle != string(caOnDisk) {
+		if !isRunningOnPod() {
```

Comment:
Question:
- If we are not running inside a Pod, why do we even have this reconciler? We can simply check if we are running inside a pod when we create the controller and choose not to instantiate it, correct?

Reply:
That's true, I'll move the logic there.

Reply:
On second thought -  the `operator-sdk up local` is very useful for quick local development - the change that you're suggesting would make debugging CA problems require spinning up a pod, while the currently implementation provides a helpful message. Thoughts?

Reply:
This would be particularly useful if there is ever an error when reconciling the configmap.

Reply:
I think if you're actually working on this code you should be able to configure your environment to handle it. By default just disabling it seems like the right way to do it to me.

Reply:
Okay - I'll make the change.

Reply:
Good point! Does the configmap reconciler need to run on an upstream kubernetes distribution? If the answer is no then I think we should treat it as a configurable feature.